qiskit_nature/second_q/algorithms/excited_states_solvers/eigensolver_factories/numpy_eigensolver_factory.py

```diff
@@ -18,12 +18,19 @@
 from qiskit.algorithms.eigensolvers import Eigensolver, NumPyEigensolver
 from qiskit.utils.validation import validate_min
 
+from qiskit_nature.deprecation import DeprecatedType, warn_deprecated
 from qiskit_nature.second_q.problems.base_problem import BaseProblem
 from .eigensolver_factory import EigensolverFactory
 
 
 class NumPyEigensolverFactory(EigensolverFactory):
-    """A factory to construct a NumPyEigensolver."""
+    """DEPRECATED A factory to construct a NumPyEigensolver.
+
+    .. warning::
+
+        This class is deprecated! Please see :ref:`this guide <how-to-numpy>` for how to replace
```

Comment:
I was wondering about migration guide that shows the existing factory way and the new way for an identical problem. It could point to the howto for more explanation for the new way of course.

Reply:
I will write the migration guide for the 0.5 to 0.6 transition in a separate PR. That will also need to do some minor restructuring of the migration guides page in order to separate the 0.4->0.5 transition guides.